public/assets/ts/admin/editor.ts

```diff
@@ -40,6 +60,103 @@ class Editor {
     container.appendChild(saveButtonContainer);
   }
 
+  // open the rich text modal
+  private openRichTextModal() {
+    const modal = document.getElementById("modal");
+    modal.style.display = "block";
+  }
+
+  // close the rich text modal
+  private closeRichTextModal() {
+    const modal = document.getElementById("modal");
+    modal.style.display = "none";
+  }
+
+  /*
+  controls the rich text editor
+  - enables the button once a markdown field is clicked
+  - disables it otherwise
+  - saves the new markdown into the data once the new markdown has been saved
+  - saves and restores the expansion state of the json editor 
+  */
+  private onEditorEvent(
+    node: { field?: string; value?: string; path?: (string | number)[] },
+    event: Event
+  ) {
+    if (event instanceof PointerEvent) {
```

Comment:
nitpick: I think the function would look cleaner if rather than putting all of the logic in one if statement, we inverted the condition to return early.  Something like: 
```
if (!(event instanceof PointerEvent)) {
  return;
} 

// rest of the code...
```
at the beginning of the file.  It's totally a nitpick though so up to you